libnavigation-base/src/main/java/com/mapbox/navigation/base/formatter/DistanceFormatterOptions.kt

```diff
@@ -101,27 +111,23 @@ class DistanceFormatterOptions private constructor(
          * @param locale [Locale]
          * @return Builder
          */
-        fun locale(locale: Locale?) =
+        fun locale(locale: Locale) =
```

Comment:
looks that when a user changes location, `UnitType` should be changed as well, as it depends 🤔 
I would wrap these 2 params(Location and UnitType) in an object or change a method to 
```suggestion
        fun locale(locale: Locale, changeUnitTypeBasedOnLocation: Boolean) = // if changeUnitTypeBasedOnLocation == true -> change UnitType as well
```

Reply:
I think we should `not` do it, because we will have 2 methods which can change `unitType`:
- `fun unitType()`
- `fun locale()` 
while building options customer may have a mistake:
```kotlin
val options = DistanceFormatterOptions.Builder(context)
            .locale(Locale.JAPAN, true)
            .unitType(UnitType.IMPERIAL) // overrides unitType set at the line above ^^
            .roundingIncrement(INCREMENT_ONE_HUNDRED)
            .build
```
```kotlin
val options = DistanceFormatterOptions.Builder(context)
            .unitType(UnitType.IMPERIAL) 
            .locale(Locale.JAPAN, true) // overrides unitType set at the line above ^^ 
            .roundingIncrement(INCREMENT_ONE_HUNDRED)
            .build
```
I think it's an error-prone impl, would prefer to avoid it and have separate methods without side-effects.

Reply:
it looks non-consistent: SDK assigns under the hood `UnitType` based on location, but it's not when you change location. `Boolean` param made re-assign of `UnitType` explicitly, so developers might set `false` and `UntiType` is not re-set. I see another option like
- set both params via the builder's fun like(with this case may be we should expose `fun Locale#getUnitTypeForLocale` for external devs):
```kotlin
fun areaOptions(location: Location, unitType: UnitType): Builder = ...
```

@LukasPaczos @Guardiola31337 any thoughts?

Reply:
btw, `Undefined` type helps in such situations ^^. We use it as a default one. 
And before building options we check if it
-  is still `Undefined` -> use unitType depending on the locale
- `Metric/Imperial` -> use it
so I think it's worth to return `Undefined` type

Reply:
discussed offline, the way is to make `unitType` null be default and check it before building options.
if still null -> use default for the locale